app/controllers/application_controller.rb

```diff
@@ -27,4 +27,11 @@ def store_user_location!
     # :user is the scope we are authenticating
     store_location_for(:user, request.fullpath)
   end
+
+  def allow_sdr_updates?
```

Comment:
Can this be `ensure_update_available`?  The `?` method name idiom commonly indicates the return value is a boolean, but this is not.

Reply:
good point.  changing to `ensure_sdr_updatable`